extension-style/src/main/java/com/mapbox/maps/extension/style/sources/generated/GeoJsonSource.kt

```diff
@@ -35,25 +40,83 @@ class GeoJsonSource(builder: Builder) : Source(builder.sourceId) {
   private val workerHandler by lazy {
     Handler(workerThread.looper)
   }
+
+  private var initGeoJSON: GeoJson? = null
+  private var initData: String? = null
+
+  private var directSetterEnabled: Boolean = false
+
   private constructor(
     builder: Builder,
-    rawGeoJson: GeoJson?
+    geoJson: GeoJson?,
+    data: String?,
   ) : this(builder) {
-    rawGeoJson?.let {
-      workerHandler.post {
-        val property = it.toPropertyValue()
-        mainHandler.post {
-          setProperty(property, throwRuntimeException = false)
+    directSetterEnabled = builder.directSetterEnabled
+    if (directSetterEnabled) {
+      this.initGeoJSON = geoJson
+      this.initData = data
+    } else {
+      geoJson?.let {
+        workerHandler.post {
+          val property = it.toPropertyValue()
+          mainHandler.post {
+            setProperty(property, throwRuntimeException = false)
+          }
         }
       }
     }
   }
 
+  private fun setGeoJSON(geoJson: GeoJson) {
+    workerHandler.removeCallbacksAndMessages(null)
+    workerHandler.post {
+      if (delegate != null) {
+        delegate?.setStyleSourceData(sourceId, toGeoJsonData(geoJson))
+        initGeoJSON = null
```

Comment:
IMO it's confusing that class var is reset here, while it's not used and geoJson is passed as an argument. I'd move the = null to the place where initGeoJSON is passed